app/forms/decidim/participatory_processes/admin/participatory_process_form.rb

```diff
@@ -93,8 +99,63 @@ def processes
           @processes ||= Decidim::ParticipatoryProcess.where(organization: current_organization)
         end
 
+        def remove_emitter_image=(value)
+          self.remove_emitter = value if value
+          prepare_emitter_name
+        end
+
+        def emitter_name_image=(value)
+          @emitter_name_image = value if value
+          prepare_emitter_name
+        end
+
+        def emitter_name_select=(value)
+          @emitter_name_select = value if value
+          prepare_emitter_name
+        end
+
+        def emitter_image=(value)
+          @emitter_image = value if value
+          prepare_emitter
+        end
+
+        def emitter_select=(value)
+          blob = Decidim::ParticipatoryProcess.find(value).emitter.attachment.blob if value.present?
```

Comment:
Here the Participatory process is fetched twice, we can store it in a variable and call it from memory

```suggestion
          target_pp = Decidim::ParticipatoryProcess.find(value)
          blob = target_pp.emitter.attachment.blob if value.present?
```

Reply:
Also there is a multiple conditions, maybe you can : 

- Check if value is defined
- If yes, fetch PP and set variables

Then prepare_emitter and prepare_emitter_name